lib/core/src/Cardano/Wallet.hs

```diff
@@ -897,11 +900,15 @@ newWalletLayer tracer bp db nw tl = do
     _submitExternalTx
         :: ByteString
         -> ExceptT ErrSubmitExternalTx IO (Tx t)
-    _submitExternalTx payload = do
-        txWithWit@(tx,_) <- withExceptT ErrSubmitExternalTxDecode $
-            decodeExternalTx nw payload
-        withExceptT ErrSubmitExternalTxNetwork $ postTx nw txWithWit
-        return tx
+    _submitExternalTx payload =
+        case convertFromBase Base16 payload of
```

Comment:
It seems wrong that this is happening _here_ (a.k.a in the wallet business logic). The fact that this binary data might be hex-encoded is only relevant at the edge of the system (a.k.a API). As for addresses, witnesses and other `ByteString` in the code, we always consider them "raw" with no particular encoding. 

If a conversion should happen, it should be at the API-level only, but here, that payload should already be what is then called `validPayload` (which could be invalid though if it doesn't correspond to a signed transaction).

Reply:
corrected